Zend/zend_constants.c

```diff
@@ -360,16 +372,23 @@ ZEND_API zval *zend_get_constant_ex(zend_string *cname, zend_class_entry *scope,
 		}
 		free_alloca(lcname, use_heap);
 		if (ce) {
-			ret_constant = zend_hash_find(&ce->constants_table, constant_name);
-			if (ret_constant == NULL) {
+			zend_class_constant *c = zend_hash_find_ptr(&ce->constants_table, constant_name);
+			if (c == NULL) {
+				ret_constant = NULL;
```

Comment:
better to move this assignment a bit lower after "return NULL", in  that case this assignment is useless